core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -141,15 +139,29 @@ object ConfigEntityZNode {
     Json.encodeAsBytes(Map("version" -> 1, "config" -> config.asScala))
   }
   def decode(bytes: Array[Byte]): Option[Properties] = {
-    Json.parseBytes(bytes).map { js =>
-      val configOpt = js.asJsonObjectOption.flatMap(_.get("config").flatMap(_.asJsonObjectOption))
-      val props = new Properties()
-      configOpt.foreach(config => config.iterator.foreach { case (k, v) => props.setProperty(k, v.to[String]) })
-      props
+    val props = new Properties()
+    if (bytes != null) {
+      Json.parseBytes(bytes).map { js =>
```

Comment:
Since we are not using the return value anymore, we probably can do foreach instead of map and get rid of props in line 147.